hermes-frontend/src/test/groovy/pl/allegro/tech/hermes/frontend/publishing/DynamicThroughputLimiterTest.groovy

```diff
@@ -0,0 +1,119 @@
+package pl.allegro.tech.hermes.frontend.publishing
+
+import com.codahale.metrics.Metered
+import pl.allegro.tech.hermes.api.TopicName
+import pl.allegro.tech.hermes.frontend.publishing.handlers.DynamicThroughputLimiter
+import pl.allegro.tech.hermes.frontend.publishing.handlers.ThroughputLimiter
+import spock.lang.Specification
+
+import java.util.concurrent.ScheduledExecutorService
+
+class DynamicThroughputLimiterTest extends Specification {
+
+    def max = 10_000
+    def threshold = 8000
+    def desired = 6000
+    def idleThreshold = 0.5
+
+    double globalRate = 0.0
+    def globalMeter = [getOneMinuteRate: {globalRate}] as Metered
+
+    def topicName = new TopicName("group", "name")
+    def executor = Mock(ScheduledExecutorService)
+
+    def limiter = new DynamicThroughputLimiter(max, threshold, desired, idleThreshold, 1, globalMeter, executor)
+
+
+    def "global allow"() {
+        given:
+        globalRate = 10
+
+        expect:
+        limiter.check(topicName, meter(1000))
+    }
+
+    def "global block"() {
+        given:
+        globalRate = 12_000
+
+        when:
+        limiter.check(topicName, meter(1000))
+
+        then:
+        thrown ThroughputLimiter.GlobalQuotaViolationException
+    }
+
+    def "single abuser"() {
+        given:
+        def meter = meter(1000)
+        def topic1 = registerTopic("group1", "name1", meter)
+        def topic2 = registerTopic("group2", "name2", meter)
+
+        double abuserRate = 7000
+        def abuserMeter = [getOneMinuteRate: {abuserRate}, getFifteenMinuteRate: {abuserRate}] as Metered
+        def abuser = registerTopic("group3", "name3", abuserMeter)
+
+        limiter.run()
+
+        abuserRate = 2001
+
+        expect:
+        !checkThroughput(abuser, abuserMeter)
```

Comment:
It's not clear to me what to expect from this method without looking into it's implementation. It checks whether the throughput check passes for this given topic. How about `throughputCheckPasses` or `isThroughputCheckPassing`?